cogs/moderation.py

```diff
@@ -9,14 +9,18 @@ def __init__(self, bot, config):
         self.bot = bot
         self.config = config
 
+        # compile all patterns into a dictionary as form
+        # {channel_id: compiled_regex_pattern}
+        self.regex_patterns = {chan_id: re.compile(pattern)
+                               for chan_id, pattern in config["REGEX_CHANNELS"].items()}
+
     @commands.Cog.listener()
     async def on_message(self, message):
-        # check in the config if the message's channel
-        # is limited with a specific message format
-        pattern = self.config["REGEX_CHANNELS"].get(str(message.channel.id))
+        # check if the message's channel is limited with a specific message format
+        pattern = self.regex_patterns.get(str(message.channel.id))
         if pattern:
             # if the message doesn't match with the format, the bot deletes it
```

Comment:
First word of a comment should be capitalized (see [this](https://www.python.org/dev/peps/pep-0008/#comments)).
```suggestion
            # If the message doesn't match with the format, the bot deletes it
```